spec/html_preprocessor_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+describe PDFKit::HTMLPreprocessor do
+  describe "#process" do
+    before do
+      @precprocessor = PDFKit::HTMLPreprocessor
+      @root_url = 'http://example.com/'  # This mirrors Middleware#root_url's response
+      @protocol = 'http'
+    end
+
+    it "correctly parses relative url with single quotes" do
+      @body = %{<html><head><link href='/stylesheets/application.css' media='screen' rel='stylesheet' type='text/css' /></head><body><img alt='test' src="/test.png" /></body></html>}
+      body = @precprocessor.process @body, @root_url, @protocol
+      expect(body).to eq("<html><head><link href='http://example.com/stylesheets/application.css' media='screen' rel='stylesheet' type='text/css' /></head><body><img alt='test' src=\"http://example.com/test.png\" /></body></html>")
+    end
+
+    it "correctly parses relative url with double quotes" do
+      @body = %{<link href="/stylesheets/application.css" media="screen" rel="stylesheet" type="text/css" />}
+      body = @precprocessor.process @body, @root_url, @protocol
+      expect(body).to eq("<link href=\"http://example.com/stylesheets/application.css\" media=\"screen\" rel=\"stylesheet\" type=\"text/css\" />")
+    end
+
+    it "correctly parses relative url with double quotes" do
```

Comment:
Hm, I see this was copied from the middleware tests. I'm guessing special characters?